### ISSUE TITLE
Display quoted message above the user message

### ISSUE THREAD

**Issue by Otiel** [RocketChat/feature-requests#27]
When quoting a message using the "Quote" or "Reply" action, I think it would be better if the quoted message appeared above the user message, not below as it is done currently:

![image](https://user-images.githubusercontent.com/3851902/36475651-a21a41ec-16fb-11e8-8bba-545f81348779.png)

This was already asked [here](https://github.com/RocketChat/Rocket.Chat/issues/164#issuecomment-216991364), I'm creating this issue to trace this more easily.

**Comment by jpenna** [comment]
Configurable might be harder to implement, and this feature was requested more than 1 year ago. Maybe could just change the order for all.

Is there any reason why someone would want to see the replied message **after** the reply?

**Comment by mmickelson** [comment]
I agree with this request. Reading the quote after the reply is confusing. It's much more readable to see something like:
> Question or quoted message

Answer/reply

Configurable would be nice but I imagine the majority of people would would prefer to see the quote below the reply so I would say implement the hard-coded behavior and if it's requested to have it configurable, add another feature request specifically for that.

**Comment by ankar84** [comment]
I agree with this request.
It will be so nice to add the **optional choice** of displaying quotes in the Telegram and WhatsApp style (quoted text above, answer below)
and in the Rocket.Chat style (where the answer is located to above, and the quoted message is lower than the answer).

**Comment by zedar187** [comment]
Bump :)

**Comment by KimchaC** [comment]
From me too. Please implement.

**Comment by neomag** [comment]
Please implement

**Comment by prigal** [comment]
Indeed it's a weird pattern, it should at least be configurable imo.

Upvoted.

**Comment by svedrenne** [comment]
Please implement "Reply below" or "Quoted message above reply". Very Nice To Have

By the way @prigal How can I "upvote" ?

**Comment by kyletmiller** [comment]
thanks for a great platform! adding support for this request.

**Comment by ankar84** [comment]
@rodrigok Hello, Rodrigo!
Please take a look at that feature request.
A lot of users asked for it more then 2 years now.

**Comment by jaykobi** [comment]
Like it is now I sometimes see that there is a quote only after I read the reply.
Much better to display the quote first.

**Comment by helgator** [comment]
Does anyone comprehend the logic behind "quote after reply" and can explain it? Maybe we are all just using RC the wrong way.

**Comment by rodrigok** [comment]
@helgator no, it's wrong, we are refactoring the messages and it will be fixed. It was made like that because we used the attachments component to display the content, and attachments were only available below the message.

**Comment by ankar84** [comment]
> we are refactoring the messages and it will be fixed

Really glad to hear that! Thank you for that work!

**Comment by Feuermurmel** [comment]
If this is touched, I'd also love to see the original message truncated to e.g. a maximum of two lines. Right now, the message flow is sometimes completely disrupted when long messages are quoted. Please have a look at how [Telegram Messenger](https://telegram.org) is doing this. IMHO they have the best UI for replies, I use it a lot and I never have trouble.

**Comment by MrVonkey** [comment]
Would it be possible to temporarily add an additional CSS class to the attachment component for quotes until the message component is refactored? This would allow admins to place quotes above the messages with custom CSS, e.g. flex-direction:

```css
.message.quote {
  display:flex;
  flex-direction: column-reverse;
}
```

Not optimal, but still an improvement for readability in my opinion.

**Comment by skjardenCode** [comment]
Since beginning of April 2022, I started using RocketChat. I was very confused by quoted messages below the answer. I also vote to change this. The reading direction (flow of chat) is top to bottom, so should be the quote-answer direction.

**Comment by ankar84** [comment]
Great news - https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3883 introduces proper direction of quotes on mobile!
Kudos to mobile team!
Still waiting for Web client implementation...

**Comment by EdSantos754** [comment]
C'mon guys, this is something **really annoying** in the rocket.chat interface. It confuses people and gets in the way of the natural flow of conversations. It's been too many years. This should be prioritized over other less useful features, as it is a daily problem that our online communities face. Thanks.

**Comment by MrVonkey** [comment]
> C'mon guys, this is something **really annoying** in the rocket.chat interface. It confuses people and gets in the way of the natural flow of conversations. It's been too many years. This should be prioritized over other less useful features, as it is a daily problem that our online communities face. Thanks.

Yes absolutely, this should be prioritized. Unfortunately I couldn't find anything concerning this in the changelog for the 5.0.0 release candidate. Maybe I missed something, since there are several issues/requests related to this?

**Comment by MrVonkey** [comment]
Since version 5.0.0 there are more possibilities to customize the layout of the message component. You can try the following custom CSS in Settings > Layout to place the username / message header above the quoted message block and everything else below the quoted message block:

```css
.rcx-message-container > div {
 order: 2; 
}

.rcx-message-container > div.rcx-message-header {
  order: 0;
}

.rcx-message-container > div.rcx-attachment__content {
  order: 1;
}
```
